domain/src/main/kotlin/no/nav/su/se/bakover/domain/kontrollsamtale/Kontrollsamtale.kt

```diff
@@ -0,0 +1,81 @@
+package no.nav.su.se.bakover.domain.kontrollsamtale
+
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.endOfMonth
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.common.startOfMonth
+import no.nav.su.se.bakover.common.zoneIdOslo
+import no.nav.su.se.bakover.domain.vedtak.Vedtak
+import java.time.Clock
+import java.time.LocalDate
+import java.util.UUID
+
+data class Kontrollsamtale(
+    val id: UUID = UUID.randomUUID(),
+    val opprettet: Tidspunkt = Tidspunkt.now(),
+    val sakId: UUID,
+    val innkallingsdato: LocalDate?,
+    val status: Kontrollsamtalestatus,
+    val frist: LocalDate?
+) {
+    companion object {
+        fun fraVedtak(vedtak: Vedtak): Kontrollsamtale {
```

Comment:
For meg fremstår det kanskje mer naturlig at oppgaven med å opprette/planlegge kontrollsamtaler delegeres til `Sak` (siden det er noe som potensielt vil måtte spenne og vedlikeholdes på tvers av mange vedtak). Det ville kanskje også løst modelleringsfloken med kontrollsamtaler uten innkallingsdato/frist (ikke opprett noe kontrollsamtale/frist dersom det ikke er relevant?)

Reply:
Jeg tenkte først at dette sannsynligvis bare var relevant ved innvilgelse av ny stønadsperiode, men etter å ha resonnert litt rundt hvordan logikken for opprettelse av "neste" kontrollsamtale skjønner jeg kanskje hvorfor den ikke er typet spesifikt til vedtak for ny stønadsperiode - var tanken at denne funksjonen også skulle kalles for opprettelsen av "neste" kontrollsamtale også? Dette vil vel i så fall fort by på problemer, da vi sannsynligvis ikke ønsker å opprette nye kontrollsamtaler ved opphør, eller ei heller doble kontrollsamtaler ved revurdering (understøtter også argumentet mitt for at saken skal håndtere dette - siden den allerede kjenner til evt eksisterende kontrollsamtaler).

Reply:
Den er knyttet til `Sak` i form av at den har sakId som property, men problemet med å opprette det fra Sak (og ikke Vedtak), er at det er Vedtak som har kontroll på invilgelsesperioden. Man skal ikke opprette en kontrollsamtale når en sak opprettes, men derimot når vedtaket om innvilgelse blir gjort.

Tanken er videre at ved operasjoner som påvirker kontrollsamtale-utsendelsen også skal endre datoen. Mulig dette ikke er gjennomtenkt nok dog, så vi kan ta en ny runde her

Reply:
Tanken var vel at en sak _både_  bør kjenne til alle vedtak (og dermed alle "innvilgelsesperioder") i tillegg til alle eventuelle kontrollsamtaler som måtte eksistere fra før (sikkert nyttig hvis du har tenkt å f.eks justere datoene/fjerne/legge til eller lignende). Det er ikke snakk om at kontrollsamtalen skal opprettes sammen med saken, men at opprettelse av kontrollsamtaler er en funksjonalitet som saken tilbyr (da dette objektet kjenner til all den informasjonen som er nødvendig)

Reply:
Vi skriver om slik at det er _servicen_ som håndterer opprettelse av kontrollsamtaler. Da slipper vi en konstruktør som tilsynelatende er veldig koblet mot et vedtak. Kommer en ny commit straks, så kan vi evt snakke om hvordan det ser ut da